container/docker/factory.go

```diff
@@ -72,11 +79,21 @@ var (
 
 func RootDir() string {
 	dockerRootDirOnce.Do(func() {
-		status, err := Status()
-		if err == nil && status.RootDir != "" {
-			dockerRootDir = status.RootDir
-		} else {
-			dockerRootDir = *dockerRootDirFlag
+		var remaining, initTime, succeed = GetRootDirMaxTime, GetRootDirInitTime, false
```

Comment:
This is pretty hard to read... Can we just do:

```
for i  := 0; i < rootDirRetries; i++ {
    status, err := Status()
    if err == nil && status.RootDir != "" {
        dockerRootDir = status.RootDir
        break
    } else {
        time.Sleep(rootDirRetryPeriod)
    }
}
if dockerRootDir == "" {
    dockerRootDir = *dockerRootDirFlag
}

Reply:
Thanks for your review!
My solution is just sleep for different milliseconds each time if get root failed, and each time is twice as much as last time. But it is really hard to read than this...
Shall we wait for some more reviewers to comment, and i will correct this if most of us agree with this easier way.
Thanks again for your review!

Reply:
I think for a while, it is really not necessary to make this retry complicated, i will correct it, thanks for your advice~

Reply:
I have update the code, thanks your advice ~

Reply:
yeah, sadly I'm the only reviewer active for this repository.